prov/gni/src/gnix_mr.c

```diff
@@ -61,11 +112,123 @@ static inline int64_t __sign_extend(uint64_t val, int len)
 	return r;
 }
 
+/**
+ * Key comparison function for gnix memory registration caches
+ *
+ * @param[in] x key to be inserted or found
+ * @param[in] y key to be compared against
+ *
+ * @return    -1 if it should be positioned at the left, 0 if the same,
+ *             1 otherwise
+ */
+static inline int __mr_cache_key_comp(
```

Comment:
From this specific function, yes. From the other functions, that's debatable. I'd prefer to be explicit in this case as I would like the function calls to be replaced with the actual code itself.